projects/RabbitMQ.Client/client/impl/SocketFrameHandler.cs

```diff
@@ -262,51 +267,105 @@ public void SendHeader()
 
         public void Write(OutgoingFrame frame)
         {
-            _channelWriter.TryWrite(frame);
+            LockToken token = default;
+            try
+            {
+                token = _singleWriterMutex.TryWait(WaitOptions.NoDelay);
+                if (!token.Success)
+                {
+                    // Didn't get the lock immediately, let's backlog the write and start the backlog writer task.
+                    if (_channelWriter.TryWrite(frame))
+                    {
+                        if (_writerTask == null || _writerTask.Status == TaskStatus.RanToCompletion)
+                        {
+                            _writerTask = Task.Run(WriteLoop, CancellationToken.None);
```

Comment:
This (although unlikely) could lead to a race condition where two Tasks are created.

Reply:
Yeah, I have yet to clean that up a bit, thanks for pointing it out :)